pkg/operator/configobserver/apiserver/observe_cors.go

```diff
@@ -10,33 +10,50 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-var clusterDefaultCORSAllowedOrigins = []string{
-	`//127\.0\.0\.1(:|$)`,
-	`//localhost(:|$)`,
+var (
+	clusterDefaultCORSAllowedOrigins = []string{
+		`//127\.0\.0\.1(:|$)`,
+		`//localhost(:|$)`,
+	}
+
+	corsAllowedOriginsPath = map[bool][]string{
+		false: {"corsAllowedOrigins"},
+		true:  {"apiServerArguments", "cors-allowed-origins"},
+	}
+)
+
+type corsObserver struct {
+	corsAllowedOriginsPath []string
 }
 
-// ObserveAdditionalCORSAllowedOrigins observes the additionalCORSAllowedOrigins field
-// of the APIServer resource
-func ObserveAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	const corsAllowedOriginsPath = "corsAllowedOrigins"
+// NewAdditionalCORSAllowedOriginsObserver creates an observer for the additionalCORSAllowedOrigins
+// field of the APIServer resource.
+// Set useAPIServerArguments to true if you want to use the apiServerArguments field instead
+func NewAdditionalCORSAllowedOriginsObserver(useAPIServerArguments bool) configobserver.ObserveConfigFunc {
+	observer := &corsObserver{
+		corsAllowedOriginsPath: corsAllowedOriginsPath[useAPIServerArguments],
+	}
+	return observer.observeAdditionalCORSAllowedOrigins
+}
 
+func (o *corsObserver) observeAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
 	lister := genericListers.(APIServerLister)
 	errs := []error{}
 	defaultConfig := map[string]interface{}{}
-	if err := unstructured.SetNestedStringSlice(defaultConfig, clusterDefaultCORSAllowedOrigins, corsAllowedOriginsPath); err != nil {
+	if err := unstructured.SetNestedStringSlice(defaultConfig, clusterDefaultCORSAllowedOrigins, o.corsAllowedOriginsPath...); err != nil {
 		// this should not happen
 		return defaultConfig, append(errs, err)
 	}
 
 	// grab the current CORS origins to later check whether they were updated
-	currentCORSAllowedOrigins, _, err := unstructured.NestedStringSlice(existingConfig, corsAllowedOriginsPath)
+	currentCORSAllowedOrigins, _, err := unstructured.NestedStringSlice(existingConfig, o.corsAllowedOriginsPath...)
 	if err != nil {
 		return defaultConfig, append(errs, err)
 	}
 	currentCORSSet := sets.NewString(currentCORSAllowedOrigins...)
 	currentCORSSet.Insert(clusterDefaultCORSAllowedOrigins...)
 
-	observedConfig := map[string]interface{}{}
+	observedConfig := prepareEmptyInterfaceForPath(o.corsAllowedOriginsPath...)
```

Comment:
even start with `map[string]interface{}{}`. No need to prepare anything. `SetNestedStringSlice` will create the path on the way.

Reply:
I could have sworn that that was failing for me with non-nil asserts before, perhaps I was doing something wrong because now they are not. Thanks